### DISCUSSION TITLE
HttpOnly flag not set in KEYCLOAK_SESSION & KEYCLOAK_SESSION_LEGACY Cookie

### DISCUSSION THREAD

**Discussion by Gokulamaanickam** [keycloak/keycloak#32714] [Q&A]
From this discussion [Pull-16747](https://github.com/keycloak/keycloak/pull/16747) , we could see that currently both the cookies dont have httpOnly flag set.
I have come to understand that this might make it possible to inject JavaScript into the application, potentially stealing session cookies and performing an account takeover.
What are the ways in which Keycloak guards against these kind of attacks ?

**Comment by jonkoops** [comment]
The omission of the `HttpOnly` attribute does not mean that you can 'inject' JavaScript, it only means that this cookie is exposed to JavaScript. We need this in order to read out the session in various places to identify if the user has an active session. The information contained within this cookie does not include any sensitive information, and is not exposed to third parties in an insecure manner.

I see no way this could result in hijacking a user account.  If you believe you found a security vulnerability that is actually exploitable then please follow our [security policy](https://www.keycloak.org/security).
